src/apply_manual_codes.py

```diff
@@ -0,0 +1,167 @@
+import time
+from os import path
+import sys
+
+from core_data_modules.cleaners import Codes
+from core_data_modules.cleaners.cleaning_utils import CleaningUtils
+from core_data_modules.traced_data import Metadata
+from core_data_modules.traced_data.io import TracedDataCodaV2IO
+from core_data_modules.util import TimeUtils
+
+from src.lib import PipelineConfiguration, CodeSchemes
+
+class ApplyManualCodes(object):
+    @classmethod
+    def apply_manual_codes(cls, user, data, coda_input_dir):
+        # Merge manually coded radio show files into the cleaned dataset
+        for plan in PipelineConfiguration.RQA_CODING_PLANS:
+            rqa_messages = [td for td in data if plan.raw_field in td]
+            coda_input_path = path.join(coda_input_dir, plan.coda_filename)
+    
+            f = None
+            try:
+                if path.exists(coda_input_path):
+                    f = open(coda_input_path, "r")
+                TracedDataCodaV2IO.import_coda_2_to_traced_data_iterable_multi_coded(
+                    user, rqa_messages, plan.id_field, {plan.coded_field: plan.code_scheme}, f)
+                
+                if plan.binary_code_scheme is not None:
+                    if f is not None:
+                        f.seek(0)
+                    TracedDataCodaV2IO.import_coda_2_to_traced_data_iterable(
+                        user, rqa_messages, plan.id_field, {plan.binary_coded_field: plan.binary_code_scheme}, f)
+            finally:
+                if f is not None:
+                    f.close()
+
+        # Merge manually coded follow up surveys into the cleaned dataset
+        for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+            follow_up_messages = [td for td in data if plan.raw_field in td]
+            coda_input_path = path.join(coda_input_dir, plan.coda_filename)
+    
+            f = None
+            try:
+                if path.exists(coda_input_path):
+                    f = open(coda_input_path, "r")
+                TracedDataCodaV2IO.import_coda_2_to_traced_data_iterable_multi_coded(
+                    user, follow_up_messages, plan.id_field, {plan.coded_field: plan.code_scheme}, f)
+            
+            finally:
+                if f is not None:
+                    f.close()
+        
+        # Label the RQA & surveys for which there is no response yet as TRUE MISSING
```

Comment:
1. You need to stress that this is the follow-up surveys (just saying 'surveys' suggests that this is being applied to the demogs too)
2. You're also labelling NCs here

Reply:
similar loops merged thanks!